src/sentry/snuba/metrics_layer/query.py

```diff
@@ -0,0 +1,91 @@
+from typing import Sequence
+
+from snuba_sdk import Request
+from snuba_sdk.metrics_query import MetricsQuery
+
+from sentry.models import Project
+from sentry.sentry_metrics.utils import resolve_weak, string_to_use_case_id
+from sentry.snuba.metrics.fields.base import _get_entity_of_metric_mri, org_id_from_projects
+from sentry.snuba.metrics.naming_layer.mapping import get_mri, get_public_name_from_mri
+from sentry.snuba.metrics.utils import to_intervals
+
+
+def run_query(request: Request) -> None:
+    """
+    Entrypoint for executing a metrics query in Snuba.
+
+    First iteration:
+    The purpose of this function is to eventually replace datasource.py::get_series().
+    As a first iteration, this function will only support single timeseries metric queries.
+    This means that for now, other queries such as total, formula, or meta queries
+    will not be supported. Additionally, the first iteration will only support
+    querying raw metrics (no derived). This means that each call to this function will only
+    resolve into a single request (and single entity) to the Snuba API.
+    """
+    metrics_query = request.query
+    assert isinstance(metrics_query, MetricsQuery)
+
+    assert len(metrics_query.scope.org_ids) == 1  # Initially only allow 1 org id
+    organization_id = metrics_query.scope.org_ids[0]
+    tenant_ids = request.tenant_ids or {"organization_id": organization_id}
+    if "use_case_id" not in tenant_ids:
+        tenant_ids["use_case_id"] = metrics_query.scope.use_case_id
```

Comment:
Making sure we're not sending `"use_case_id": None` in `tenant_ids`

```suggestion
    if "use_case_id" not in tenant_ids and metrics_query.scope.use_case_id is not None:
        tenant_ids["use_case_id"] = metrics_query.scope.use_case_id
```